### PR TITLE
Add the postgres to HoH operator

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>

Provide the following information/secret to authorize the HoH operator to access the database(Either automatically created by the HoH or an external database)
```bash
kubectl create secret generic "postgresql-secret" -n "open-cluster-management" \
    --from-literal=host="*" \
    --from-literal=port="5432" \
    --from-literal=user="*" \
    --from-literal=password="*" \
    --from-literal=database="*"
```
Two fields[`process_uri`, `transport_uri`] are added to the secret when the HoH operator successfully initializes the database.
```
» oc get secret postgresql-secret -oyaml
apiVersion: v1
data:
  database: *
  host: *
  password: *
  port: *
  process_uri: *
  transport_uri: *
  user: *
kind: Secret
metadata:
  creationTimestamp: "2022-08-03T09:07:28Z"
  name: postgresql-secret
  namespace: open-cluster-management
  resourceVersion: "6679"
  uid: 4cccf1d7-b066-453f-99d7-b4054217db43
type: Opaque
```
NOTE: The secret name `postgresql-secret` can be any value, as long as it is the same as postgresql.name in the HoH CR.

### CODE REVIEW

**Comment by yanmxa** [comment]
Reference: https://github.com/stolostron/backlog/issues/24506
Fixed: https://github.com/stolostron/backlog/issues/24726

File: operator/pkg/controllers/hubofhubs/config_controller.go [hunk 1]
```diff
@@ -49,6 +50,8 @@ var fs embed.FS
 
 var isLeafHubControllerRunnning = false
 
+const NAMESPACE = "open-cluster-management"
```
Comment (morvencao) [inline operator/pkg/controllers/hubofhubs/config_controller.go:53]: nit: reuse constant in `constants` package.

File: operator/pkg/controllers/hubofhubs/database.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"k8s.io/apimachinery/pkg/types"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/jackc/pgx/v4"
+	hubofhubsv1alpha1 "github.com/stolostron/hub-of-hubs/operator/apis/hubofhubs/v1alpha1"
+)
+
+const (
+	PROCESS_URI_KEY   = "process_uri"
+	TRANSPORT_URI_KEY = "transport_uri"
```
Comment (morvencao) [inline operator/pkg/controllers/hubofhubs/database.go:21]: nit: I prefer to put all constants in `constants` package.

**Review by morvencao — COMMENTED** [review]
Well done! @yanmxa 
except some small nits, overall looks goods to me!
/LGTM
/Approve

**Comment by morvencao** [comment]
ref: https://github.com/stolostron/backlog/issues/24609

**Pull request merged**